src/Files.App/Data/Contracts/IWindowsRecycleBinService.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) 2024 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+namespace Files.App.Data.Contracts
+{
+	/// <summary>
+	/// Provides service for Recycle Bin on Windows.
+	/// </summary>
+	public interface IWindowsRecycleBinService
+	{
+		/// <summary>
+		/// Gets the watcher of Recycle Bin folder.
+		/// </summary>
+		RecycleBinWatcher Watcher { get; }
+
+		/// <summary>
+		/// Gets all Recycle Bin shell folders.
+		/// </summary>
+		/// <returns>A collection of Recycle Bin shell folders.</returns>
+		Task<List<ShellFileItem>> GetAllRecycleBinFoldersAsync();
+
+		/// <summary>
+		/// Gets the info of Recycle Bin.
+		/// </summary>
+		/// <param name="drive">The drive letter, Recycle Bin of which you want.</param>
+		/// <returns></returns>
+		(bool HasRecycleBin, long NumItems, long BinSize) QueryRecycleBin(string drive = "");
+
+		/// <summary>
+		/// Gets the used size of Recycle Bin.
+		/// </summary>
+		/// <returns></returns>
+		ulong GetSize();
+
+		/// <summary>
+		/// Gets the value that indicates whether Recycle Bin folder has item(s).
+		/// </summary>
+		/// <returns></returns>
+		bool HasItems();
+
+		/// <summary>
+		/// Gets the file or folder specified is already moved to Recycle Bin.
+		/// </summary>
+		/// <param name="path">The path that indicates to a file or folder.</param>
+		/// <returns>True if the file or path is recycled; otherwise, false.</returns>
+		bool IsRecycled(string? path);
```

Comment:
```suggestion
		bool IsPathUnderRecycleBin(string? path);
```

Reply:
why that should be that long?

Reply:
`IsRecycled` is vague

Reply:
While this method is in Windows**RecycleBin**Service?

Reply:
`IsRecycled` sounds likes it's fully deleted. Another option is `IsInRecycleBin`.

Reply:
The same problem persists as above.
Recycle Bin is the name only on Windows.
If you think platform independence, we should avoid that naming. IsInBin or something sounds better to me.

---

Sorry for taking time for this tiny things but I believe discussing naming is really important to me as I try to keep overlooking Files architecture to avoid ambiguity or misunderstanding.
Thanks!